src/rest_framework_reactive/observer.py

```diff
@@ -125,274 +124,325 @@ def _get_logging_id(self):
             self._request.viewset_method,
         )
 
-    def evaluate(self, return_full=True, return_emitted=False):
-        """Evaluate the query observer.
+    def _warning(self, msg, duration=None, results=None):
+        """Log warnings."""
+        logger.warning(
+            "{} ({})".format(msg, self._get_logging_id()),
+            extra=self._get_logging_extra(duration=duration, results=results),
+        )
 
-        :param return_full: True if the full set of rows should be returned
-        :param return_emitted: True if the emitted diffs should be returned
-        """
+    def subscribe(self, session_id):
+        """Initialize observer and register subscriber.
 
+        :param session_id: Subscriber's session identifier
+        """
         try:
-            settings = get_queryobserver_settings()
+            change_detection = self._meta.change_detection
+            if change_detection not in [
+                Options.CHANGE_DETECTION_PUSH,
+                Options.CHANGE_DETECTION_POLL,
+            ]:
+                raise NotImplementedError(
+                    "Change detection mechanism '{}' not implemented.".format(
+                        change_detection
+                    )
+                )
+
+            tables = set()
+            with QueryInterceptor().intercept(tables):
+                viewset_results = self._viewset_results()
 
+            if not tables and change_detection == Options.CHANGE_DETECTION_PUSH:
+                self._warning("Will not create PUSH observer without dependencies")
+                return viewset_results
+
+            poll_interval = (
+                self._meta.poll_interval
+                if change_detection == Options.CHANGE_DETECTION_POLL
+                else None
+            )
+
+            # Subscribe to observer in a single query. First, create an
+            # observer, then create a subscriber, and finally subscribe to
+            # the observer. If already subscribed, ignore the conflict.
             for retry in range(MAX_INTEGRITY_ERROR_RETRIES):
+                is_subscribed = False
+                cursor = connection.cursor()
                 try:
-                    with transaction.atomic():
-                        # Obtain the observer state and lock it. This prevents an observer from being
-                        # processed in parallel from multiple different workers.
-                        observer, _ = models.Observer.objects.select_for_update().get_or_create(
-                            id=self.id,
-                            defaults={'request': pickle.dumps(self._request)},
+                    cursor.execute(
+                        """
+                        WITH inserted_observer AS (
+                            INSERT into rest_framework_reactive_observer ("id", "request", "poll_interval")
```

Comment:
Use `QueryObserver._meta.db_table` to get the table name. The same goes for all tables in this query.